bugwarrior/services/trello.py

```diff
@@ -0,0 +1,159 @@
+        """
+        Return extra config options to be passed to the TrelloIssue class
+        """
+        return {
+            'import_labels_as_tags':
+            self.config_get_default('import_labels_as_tags', False, asbool),
+            'label_template':
+            self.config_get_default('label_template', DEFAULT_LABEL_TEMPLATE),
+            }
+
+    def issues(self):
+        """
+        Returns a list of dicts representing issues from a remote service.
+        """
+        # First, we get the board name
+        board_id = self.config_get('board')
+        board = self.api_request(
+            "/1/boards/{board_id}".format(board_id=board_id), fields='name')
+        for lst in self.get_lists(board_id):
+            listextra = dict(boardname=board['name'], listname=lst['name'])
+            for card in self.get_cards(lst['id']):
+                yield self.get_issue_for_record(card, extra=listextra)
+
+    def get_lists(self, board):
+        """
+        Returns a list of the filtered lists for the given board
+        This filters the trello lists according to the configuration values of
+        trello.include_lists and trello.exclude_lists.
+        """
+        lists = self.api_request(
+            "/1/boards/{board_id}/lists/open".format(board_id=board),
+            fields='name')
+        try:
+            include_lists = self.config_get_list('include_lists')
+            lists = [l for l in lists if l['name'] in include_lists]
+        except NoOptionError:
+            pass
+        try:
+            exclude_lists = self.config_get_list('exclude_lists')
+            lists = [l for l in lists if l['name'] not in exclude_lists]
+        except NoOptionError:
+            pass
+        return lists
+
+    def get_cards(self, list_id):
+        """ Returns an iterator for the cards in a given list, filtered
+        according to configuration value trello.only_if_member """
+        params = {'fields': 'name,idShort,shortLink,shortUrl,url,labels'}
+        member = self.config_get_default('only_if_member', None)
+        if member is not None:
+            params['members'] = 'true'
+            params['member_fields'] = 'username'
+        cards = self.api_request(
+            "/1/lists/{list_id}/cards/open".format(list_id=list_id),
+            **params)
+        for card in cards:
+            if (member is None
+                    or member in [m['username'] for m in card['members']]):
+                yield card
+
+    def api_request(self, url, **params):
+        """
+        Make a trello API request. This takes an absolute url (without protocol
+        and host) and a list of argumnets and return a GET request with the
+        key and token from the configuration
+        """
+        params['key'] = self.config_get('api_key'),
+        params['token'] = self.config_get('token'),
+        url = "https://api.trello.com" + url
+        return requests.get(url, params=params).json()
```

Comment:
Could you add the `ServiceClient` mixin to this class and use it's `json_response` method here? It just keeps the IO error handling consistent across services.